test/ussd_popi_user_data.test.js

```diff
@@ -0,0 +1,630 @@
+                        state: 'state_all_options_view',
+                        reply: [
+                            'What would you like to do?',
+                            '1. See my personal info',
+                            '2. Change my info',
+                            '3. Request to delete my info'
+                        ].join('\n')
+                    })
+                    .check(function(api) {
+                        utils.check_fixtures_used(api, [51, 54, 171]);
+                    })
+                    .run();
+                });
+
+                describe('user selects to view details', function() {
+                    it('should go to state_view', function() {
+                        return tester
+                        .setup.user.addr('27820001002')
+                        .inputs(
+                            {session_event: 'new'}
+                            , '1' // pick option 1
+                        )
+
+                        .check.interaction({
+                            state: 'state_view',
+                            reply: [
+                                'Personal info:',
+                                'Phone number: +27820001002',
+                                'ID number: 5101025009086',
+                                'Date of birth: 1951-01-02',
+                                'Language: ',
+                                '1. Send to me by sms',
+                                '2. Back'
+
+                            ].join('\n')
+                        })
+                        .check(function(api) {
+                            utils.check_fixtures_used(api, [51, 54, 171]);
+                        })
+                        .run();
+                    });
+
+                    describe('user selects to sms details', function() {
+                        it('should go to state_view_sms', function() {
+                            return tester
+                            .setup.user.addr('27820001002')
+                            .inputs(
+                                {session_event: 'new'}
+                                , '1' // pick option 1
+                                , '1' // pick option 1
+                            )
+                            .check.interaction({
+                                state: 'state_view_sms',
+                                reply: 'An SMS has been sent to your number ' +
+                                    'containing your personal information ' +
+                                    'stored by MomConnect.'
+                            })
+                            .check(function(api) {
+                                utils.check_fixtures_used(api, [51, 54, 141, 171]);
+                            })
+                            .check.reply.ends_session()
+                            .run();
+                        }); 
+                    });
+                });
+
```

Comment:
You should probably add two tests similar to the above ones but with the messageset info in the output.